rcongui/src/components/PlayerInfo/PlayerInfo.js

```diff
@@ -0,0 +1,267 @@
+    };
+    const columns = [
+        {name: "action_type", label: "Type"},
+        {name: "reason", label: "Reason"},
+        {name: "by", label: "By"},
+        {
+            name: "time",
+            label: "Time",
+            options: {
+                customBodyRenderLite: (dataIndex) =>
+                    moment(punishments[dataIndex].time).format("ddd Do MMM HH:mm:ss"),
+            },
+        }
+    ];
+
+    const options = {
+        filter: false,
+        rowsPerPage: myRowPerPage,
+        selectableRows: "none",
+        rowsPerPageOptions: [10, 25, 50, 100, 250, 500, 1000],
+        onChangeRowsPerPage: saveRowsPerPage,
+    };
+
+    return (
+        <MUIDataTable
+            title={"Punishments"}
+            data={punishments}
+            columns={columns}
+            options={options}
+        />
+    );
+}
+
+class PlayerInfo extends React.Component {
+    static propTypes = {
+        match: PropTypes.object.isRequired,
+        location: PropTypes.object.isRequired,
+        history: PropTypes.object.isRequired
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            steam_id_64: "",
+            created: "0",
+            names: [],
+            sessions: [],
+            sessions_count: 1086,
+            total_playtime_seconds: 4495950,
+            current_playtime_seconds: 17455,
+            received_actions: [],
+            penalty_count: {
+                TEMPBAN: 0,
+                PERMABAN: 0,
+                PUNISH: 0,
+                KICK: 0
+            },
+            blacklist: {
+                is_blacklisted: false,
+                reason: "",
+                by: ""
+            },
+            flags: [],
+            watchlist: {},
+            steaminfo: {},
+            perma: false,
+            temp: false,
+            vip: false,
+            loaded: false
+        }
+    }
+
+    /**
+     * fetch bans that currently affect the steamId64
+     * @param steamId64
+     */
+    fetchPlayerBan(steamId64) {
+        get(`get_ban?steam_id_64=${steamId64}`)
+            .then((data) => {
+                const temp = data.result.find((ban, index) => {
+                    return ban.type === "temp";
+
+                })
+                const perma = data.result.find((ban, index) => {
+                    return ban.type === "perma";
+                })
+                if (temp !== undefined) {
+                    this.setState({temp: true})
+                }
+                if (perma !== undefined) {
+                    this.setState({perma: true})
+                }
+            })
+            .catch(handle_http_errors);
+    }
+
+    /**
+     * fetch Player data
+     * @param steamId64
+     */
+    fetchPlayer(steamId64) {
+        get(`player?steam_id_64=${steamId64}`)
+            .then((response) => showResponse(response, "get_user", false))
+            .then((data) => {
+                this.setState({
```

Comment:
you should check if `result` is actually not null, because it's always possible that there was an error, and this code would crash